AirCasting/SessionsSynchronization/Controller/SessionSynchronizationViewModel.swift

```diff
@@ -0,0 +1,20 @@
+// Created by Lunar on 09/09/2021.
+//
+
+import Foundation
+import Combine
+
+protocol SessionSynchronizationViewModel: ObservableObject {
+    var syncInProgress: Bool { get }
+}
+
+class DefaultSessionSynchronizationViewModel: SessionSynchronizationViewModel {
+    @Published public var syncInProgress: Bool = true
+    private var cancellables = [AnyCancellable]()
+    
+    init(syncSessionController: SessionSynchronizer) {
+        syncSessionController.syncInProgress.sink { [weak self] value in
```

Comment:
when you change those DispatchQueue.main.async from controller, lets add `.receive(on: DispatchQueue.main)` (before `.sink`)